src/time.cc

```diff
@@ -57,17 +62,36 @@ std::string ToString(const std::chrono::system_clock::time_point& t) {
 
 std::chrono::system_clock::time_point FromString(const std::string& s) {
   std::tm tm;
-  const char* end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:%S", &tm);
-  if (end == nullptr || end - s.c_str() != s.find('.')) {
-    // TODO
+  char* const end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:%S", &tm);
+  if (end == nullptr || !std::isdigit(*(end - 2))) {
+    // TODO: Invalid time format.
     return std::chrono::system_clock::time_point();
   }
   char* zone;
-  long ns = std::strtol(end + 1, &zone, 10);
-  if (zone <= end + 1 || *zone != 'Z' || *(zone+1) != '\0') {
-    // TODO
+  if (*end == '.') {
+    (void)std::strtol(end + 1, &zone, 10);
+    if (zone <= end + 1) {
+      // TODO: Missing nanoseconds.
+      return std::chrono::system_clock::time_point();
+    }
+  } else {
+    zone = end;
+  }
+  if (*zone != 'Z' || *(zone+1) != '\0') {
+    // TODO: Invalid timezone.
+    return std::chrono::system_clock::time_point();
+  }
+  char* d_end;
+  double seconds = std::strtod(end - 2, &d_end);
+  if (d_end != zone) {
+    // TODO: Internal error.
     return std::chrono::system_clock::time_point();
   }
+  static_assert(sizeof(long) == 8, "long is too small");
```

Comment:
When would this happen?

Reply:
For example with a 32-bit compiler. I'd rather fail compilation if someone ever tries to build for Amazon's 32-bit AMI or Windows than silently overflow.